recipes/faiss/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import can_run, check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
+from conan.tools.scm import Version
+
+
+class FaissRecipe(ConanFile):
+    name = "faiss"
+    package_type = "library"
+
+    # Optional metadata
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Faiss - a library for efficient similarity search and clustering of dense vectors"
+    topics = ("approximate-nearest-neighbor", "ann", "anns")
+    homepage = "https://github.com/facebookresearch/faiss"
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_opt_generic": [True, False],
+        "with_opt_avx2": [True, False],
+        "with_opt_avx512": [True, False],
+        "with_gpu": [True, False],
+        "with_c_api": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "with_opt_generic": False,
+        "with_opt_avx2": True,
+        "with_opt_avx512": False,
+        "with_gpu": False,
+        "with_c_api": False,
+        # TODO: current openblas recipe does not export all below options yet
+        # "openblas:use_openmp": True,
+        # "openblas:use_thread": False,
+        # "openblas:use_locking": False,
+        # "openblas:num_threads": 512,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder,
+            strip_root=True)
+        apply_conandata_patches(self)
+
+    @property
+    def _enable_testing(self):
+        return not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+    def config_options(self):
+        if self.settings.arch != "x86_64":
+            del self.options.with_opt_avx2
+            del self.options.with_opt_avx512
+        if Version(self.version) < "1.8.0":
+            del self.options.with_opt_avx512
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("openblas/0.3.27")
+        self.test_requires("gtest/1.12.1")
```

Comment:
`self.test_requires()` be placed under `build_requirements()` instead.

Besides that, following the value of the `tools.build:skip_test` config option on CCI is discouraged as it can lead to unexpected test failures for consumers and CCI maintainers try to avoid the additional potential burden from bug reports related to these.